common/src/unifyfs_client_rpcs.h

```diff
@@ -282,6 +283,105 @@ MERCURY_GEN_PROC(unifyfs_heartbeat_in_t,
 MERCURY_GEN_PROC(unifyfs_heartbeat_out_t, ((int32_t)(ret)))
 DECLARE_MARGO_RPC_HANDLER(unifyfs_heartbeat_rpc)
 
+/* unifyfs_node_local_extents_get_rpc (client => server)
+ *
+ * returns node local extents for a file
+ * given a global file id */
+MERCURY_GEN_STRUCT_PROC(unifyfs_index_t,
+                 ((uint32_t)(file_pos))
```

Comment:
uint32_t is too small for file/log positions and extent length. use hg_size_t or uint64_t

Reply:
You're not tracking the client_id for the returned extents. The client_id is necessary to know which data log to read. With the current code, I think you have a bug where you are always reading the log for the current client, rather than the log for another client on the same node. There is additional work that will need to be done to setup the logio state for other clients and use it when reading extents from those clients. I think this also means you can't use the existing `extents` seg_tree in the client's file metadata to store the node-local extents, since `seg_tree` has no notion of other clients.

Reply:
I fixed this check my comment below